R/model.R

```diff
@@ -1387,14 +1306,50 @@ diagnose_method <- function(data = NULL,
     output_dir = output_dir,
     output_basename = output_basename
   )
-  cmdstan_procs <- CmdStanProcs$new(
-    num_procs = 1,
-    show_stdout_messages = FALSE,
-    show_stderr_messages = TRUE
-  )
-  runset <- CmdStanRun$new(args = cmdstan_args, procs = cmdstan_procs)
+  runset <- CmdStanRun$new(args, procs)
   runset$run_cmdstan()
 
   CmdStanDiagnose$new(runset)
 }
 CmdStanModel$set("public", name = "diagnose", value = diagnose_method)
+
+
+
+# internal ----------------------------------------------------------------
+
+assert_valid_opencl <- function(opencl_ids, cpp_options) {
+  if (is.null(cpp_options[["stan_opencl"]])
+      && !is.null(opencl_ids)) {
+    stop("'opencl_ids' is set but the model was not compiled with for use with OpenCL.",
+         "\nRecompile the model with 'cpp_options = list(stan_opencl = TRUE)'",
+         call. = FALSE)
+  }
+  invisible(opencl_ids)
+}
+
+assert_valid_threads <- function(threads, cpp_options, multiple_chains = FALSE) {
+  threads_arg <- if (multiple_chains) "threads_per_chain" else "threads"
+  checkmate::assert_integerish(threads, .var.name = threads_arg,
+                               null.ok = TRUE, lower = 1, len = 1)
```

Comment:
added this function to avoid many lines of code in each model fitting method that are basically the same except for the argument name (`threads` vs `threads_per_chain`) and I moved the checkmate assertion in each of the methods in here now